sklearn/ensemble/_forest.py

```diff
@@ -743,7 +743,7 @@ def __init__(self,
             warm_start=warm_start,
             max_samples=max_samples)
 
-    def predict(self, X):
+    def predict(self, X, check_input=True, parallel_predict=True):
```

Comment:
I would do this with context managers and to not add public parameters.

eg using:
```
with parallel_backend('threading', n_jobs=1):
     clf.predict(X)
```

and we should be able to do:

```
with sklearn.config_context(check_input=True):
     ....
```

Reply:
DecisionTreeRegressor still has a parameter check_input=False (https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeRegressor.html#sklearn.tree.DecisionTreeRegressor.predict). Which one should be used when check_input values are not the same?